roborazzi-compose/src/main/java/com/github/takahirom/roborazzi/RoborazziComposePreview.kt

```diff
@@ -0,0 +1,123 @@
+package com.github.takahirom.roborazzi
+
+import android.app.Activity
+import android.graphics.Color
+import android.view.ViewGroup
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewRootForTest
+import androidx.compose.ui.unit.dp
+import androidx.test.core.app.ActivityScenario
+import org.robolectric.Shadows
+import org.robolectric.shadows.ShadowDisplay
+import java.io.File
+import kotlin.math.roundToInt
+
+fun captureSizedRoboImage(
+  widthDp: Int,
+  heightDp: Int,
+  showBackground: Boolean,
+  backgroundColor: Long,
+  filePath: String,
+  roborazziOptions: RoborazziOptions = provideRoborazziContext().options,
+  content: @Composable () -> Unit,
+) {
+  if (!roborazziOptions.taskType.isEnabled()) return
+  registerRoborazziActivityToRobolectricIfNeeded(ComponentActivity::class.java)
```

Comment:
In my private project I had a custom Preview with heightDp = 30 and widthDp= 30 nad it crashed with the RobolectricTransparentActivity (actually, it would make sense to add tests for such cases)

That is because it has `android:windowActionBar = false`. The windowActionBar seems to be 56dp, so I assume that any height or width lower than that would make test crash. Moreover, I observed that Previews that set very high widthDp or heightDP values e.g. widthDp = 2000 and heightDp = 2000, had the height cut down by 56 dp...

That does not happen by directly using the ComponentActivity. Since I am not sure why the RobolectricTransparentActivity has `android:windowActionBar = false`, I am using ComponentActivity instead.

Reply:
Thank you for your information. We set windowActionBar to false because the activity has a shadow for the ActionBar. Do you know if the shadow is not shown with ComponentActivity?
https://github.com/takahirom/roborazzi/pull/283

Reply:
I've just added the Preview with widthDp = 30 and heightDp = 30 and it is crashing although I am using the ComponentActivity diectly... in my project it does not crash and renders properly... it only crashes if I use the RobolectricTransparentActivity... I think it might be that I am using the latest version coming with androidx.compose:compose-bom:2024.11.00 and Roborazzi uses another one where it does not work?

Reply:
Thanks. Do you have a test we can use to reproduce the issue? Is the failing test in this PR related to that problem?

Reply:
I'm not sure, but the hieght has become zero. This is only my guess, but:
1. Setting Activity.setDisplaySize
2. Set action bar size to 83
3. Set ViewGroup padding to 83
4. The size for layout become zero


<img width="696" alt="image" src="https://github.com/user-attachments/assets/4a1e537d-a7f5-4b26-9dca-38c47ae14187">
<img width="1071" alt="image" src="https://github.com/user-attachments/assets/ea0a8126-6328-4e7a-b21a-86ef4fa0bda3">
<img width="1027" alt="image" src="https://github.com/user-attachments/assets/41b30149-ba98-4763-a6ac-89ceea795df9">
<img width="851" alt="image" src="https://github.com/user-attachments/assets/de9b74bc-a4e9-46b7-bbbc-2b5a25b8af95">

Reply:
I think you can debug this issue by adding breakpoint on AbstractComposeView.internalOnMeasure

Reply:
I took a quick look, and it seems that the `android:windowActionBar` isn't applied because we can't merge themes in tests. The ActionBar was created, and it takes a margin. I've switched to using system `Theme_Translucent_NoTitleBar_Fullscreen`, and it works. I'm not sure if there is a better theme available, but this is a step forward.

<img width="1266" alt="image" src="https://github.com/user-attachments/assets/11ef373f-e7da-4b4f-9465-0e604353e7cb">

https://github.com/takahirom/roborazzi/pull/576/commits/b4ff8cfab7e5460607569b24efd15260bdd41e62

Reply:
Awesome! Once that works, I’ll try to clean this up and create a non-draft PR when I find the time.

thanks! I highly appreciate it 🙏